### ISSUE TITLE
Installation fails in pipenv

### ISSUE THREAD

**Issue by iainelder** [wandera/1password-client#25]
I'm keen to try this client as it seems the most promising for writing to 1Password from Python, but I'm falling at the first hurdle.

The full output from my installation attempt is below.

What is the installer trying to do in /usr/local/bin?

```text
  error: [Errno 1] Operation not permitted: '/usr/local/bin/'
```

Full output:

```
$ pipenv install 1password
Installing 1password...
Error:  An error occurred while installing 1password!
Error text: Collecting 1password
  Downloading 1password-0.5.2.tar.gz (11 kB)
Collecting wget>=3.2
  Using cached wget-3.2-py3-none-any.whl
Collecting pyyaml>=5.4
  Downloading PyYAML-5.4.1-cp38-cp38-manylinux1_x86_64.whl (662 kB)
Collecting pycryptodome>=3.9.7
  Downloading pycryptodome-3.10.1-cp35-abi3-manylinux2010_x86_64.whl (1.9 MB)
Requirement already satisfied: pexpect>=4.7.0 in /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/lib/python3.8/site-packages (from 1password->-r /tmp/pipenv-8vspyspb-requirements/pipenv-6kgog5vn-requirement.txt (line 1)) (4.8.0)
Requirement already satisfied: ptyprocess>=0.5 in /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/lib/python3.8/site-packages (from pexpect>=4.7.0->1password->-r /tmp/pipenv-8vspyspb-requirements/pipenv-6kgog5vn-requirement.txt (line 1)) (0.7.0)
Building wheels for collected packages: 1password
  Building wheel for 1password (setup.py): started
  Building wheel for 1password (setup.py): finished with status 'error'
  Running setup.py clean for 1password
Failed to build 1password
Installing collected packages: wget, pyyaml, pycryptodome, 1password
    Running setup.py install for 1password: started
    Running setup.py install for 1password: finished with status 'error'

  ERROR: Command errored out with exit status 1:
   command: /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3d2snmwe/1password_a611f98de08c45ec945922e778103d21/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3d2snmwe/1password_a611f98de08c45ec945922e778103d21/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-u7mn39oy
       cwd: /tmp/pip-install-3d2snmwe/1password_a611f98de08c45ec945922e778103d21/
  Complete output (38 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/onepassword
  copying onepassword/exceptions.py -> build/lib/onepassword
  copying onepassword/client.py -> build/lib/onepassword
  copying onepassword/utils.py -> build/lib/onepassword
  copying onepassword/__init__.py -> build/lib/onepassword
  warning: build_py: byte-compiling is disabled, skipping.
  
  installing to build/bdist.linux-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.linux-x86_64
  creating build/bdist.linux-x86_64/wheel
  creating build/bdist.linux-x86_64/wheel/onepassword
  copying build/lib/onepassword/exceptions.py -> build/bdist.linux-x86_64/wheel/onepassword
  copying build/lib/onepassword/client.py -> build/bdist.linux-x86_64/wheel/onepassword
  copying build/lib/onepassword/utils.py -> build/bdist.linux-x86_64/wheel/onepassword
  copying build/lib/onepassword/__init__.py -> build/bdist.linux-x86_64/wheel/onepassword
  warning: install_lib: byte-compiling is disabled, skipping.
  
  running install_egg_info
  running egg_info
  writing 1password.egg-info/PKG-INFO
  writing dependency_links to 1password.egg-info/dependency_links.txt
  writing requirements to 1password.egg-info/requires.txt
  writing top-level names to 1password.egg-info/top_level.txt
  reading manifest file '1password.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching 'LICENSE.txt'
  adding license file 'LICENSE'
  writing manifest file '1password.egg-info/SOURCES.txt'
  Copying 1password.egg-info to build/bdist.linux-x86_64/wheel/1password-0.5.2-py3.8.egg-info
  running install_scripts
  error: [Errno 1] Operation not permitted: '/usr/local/bin/'
  ----------------------------------------
  ERROR: Failed building wheel for 1password
    ERROR: Command errored out with exit status 1:
     command: /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3d2snmwe/1password_a611f98de08c45ec945922e778103d21/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3d2snmwe/1password_a611f98de08c45ec945922e778103d21/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-0uxk119k/install-record.txt --single-version-externally-managed --compile --install-headers /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/include/site/python3.8/1password
         cwd: /tmp/pip-install-3d2snmwe/1password_a611f98de08c45ec945922e778103d21/
    Complete output (35 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib
    creating build/lib/onepassword
    copying onepassword/exceptions.py -> build/lib/onepassword
    copying onepassword/client.py -> build/lib/onepassword
    copying onepassword/utils.py -> build/lib/onepassword
    copying onepassword/__init__.py -> build/lib/onepassword
    warning: build_py: byte-compiling is disabled, skipping.
    
    running install_lib
    creating /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/lib/python3.8/site-packages/onepassword
    copying build/lib/onepassword/exceptions.py -> /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/lib/python3.8/site-packages/onepassword
    copying build/lib/onepassword/client.py -> /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/lib/python3.8/site-packages/onepassword
    copying build/lib/onepassword/utils.py -> /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/lib/python3.8/site-packages/onepassword
    copying build/lib/onepassword/__init__.py -> /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/lib/python3.8/site-packages/onepassword
    warning: install_lib: byte-compiling is disabled, skipping.
    
    running install_egg_info
    running egg_info
    writing 1password.egg-info/PKG-INFO
    writing dependency_links to 1password.egg-info/dependency_links.txt
    writing requirements to 1password.egg-info/requires.txt
    writing top-level names to 1password.egg-info/top_level.txt
    reading manifest file '1password.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching 'LICENSE.txt'
    adding license file 'LICENSE'
    writing manifest file '1password.egg-info/SOURCES.txt'
    Copying 1password.egg-info to /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/lib/python3.8/site-packages/1password-0.5.2-py3.8.egg-info
    running install_scripts
    writing list of installed files to '/tmp/pip-record-0uxk119k/install-record.txt'
    error: [Errno 1] Operation not permitted: '/usr/local/bin/'
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3d2snmwe/1password_a611f98de08c45ec945922e778103d21/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3d2snmwe/1password_a611f98de08c45ec945922e778103d21/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-0uxk119k/install-record.txt --single-version-externally-managed --compile --install-headers /home/isme/.local/share/virtualenvs/root_account_creator-eZXCdEUk/include/site/python3.8/1password Check the logs for full command output.

✘ Installation Failed 
```

**Comment by dtpryce** [comment]
Just for more details are you running M1 Mac / what's your environment setup?

**Comment by dtpryce** [comment]
And to answer your question we tried to create a full installation including the AgileBits op cli which has to be on the PATH hence it tries to write to the bin. I think for your use case we have to try to work out how to know we are in a pipenv / virtual / conda env.

**Comment by iainelder** [comment]
> Just for more details are you running M1 Mac / what's your environment setup?

Hi1 @dtpryce . I'm running on it my own laptop. The OS is Ubuntu 20.04.

> we tried to create a full installation including the AgileBits op cli which has to be on the PATH hence it tries to write to the bin

Is there any way to skip that part? The op binary is already on my path and I'd like to manage it myself.

**Comment by dtpryce** [comment]
Yeah we could add a pre installed flag or similar and then it's just installing the python client?

**Comment by iainelder** [comment]
That sounds great!

How would it work?

I just learned about pip's [per-requirement overrides](https://pip.pypa.io/en/stable/cli/pip_install/#per-requirement-overrides).

In theory it gives us a possible solution:

Option to install the 1Password CLI (the default option)

```bash
pip install 1password --install-option="--install-op-cli"
pip install 1password # same as above
```

Option to skip the 1Password CLI (the one I want):

```bash
pip install 1password --install-option="--no-install-op-cli"
```

But [pipenv doesn't support these options yet](https://github.com/pypa/pipenv/issues/2208).

And that's because [Pipfile doesn't support them either](https://github.com/pypa/pipfile/issues/115).

I don't need to use pipenv. If you know of any other Python dependency managers without this limitation I'd be happy to try them.

I think another option would be to make the 1Password CLI installation an [optional dependency](https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html).

I don't see how to use an optional dependency to subtract functionality (and even if it's doable it seems weird), so this time the default behaviour would change.

Option to install just the Python client:

```
pip install 1password
```

Option to install the Python client and a managed copy of the 1Password CLI:

```
pip install 1password[managed_cli]
```

What do you think?

**Comment by dtpryce** [comment]
Actually having checked the code again the install_op script already checks if it's installed already or not ... so unless there is something wrong in that step there is something else going on here. I am also experiencing the same issue on my machine you original reported but I have removed `op` to test so I think your PATH isn't updated or something similar.

**Comment by dtpryce** [comment]
> I don't need to use pipenv. If you know of any other Python dependency managers without this limitation I'd be happy to try them.

Haha there are many but we actually use docker images for this purpose to really isolate and at a push conda envs.

**Comment by iainelder** [comment]
I can reproduce the issue from a new directory outside my current project.

op is in the PATH in the default place. As far as I know, /usr/local/bin is always in the PATH; it doesn't depend on any changes that my Bash profile may make.

So unless pipenv or pip is ignoring my PATH, I'm not sure what's going on.

```
$ cd "$(mktemp --dir)"
$ which op
/usr/local/bin/op
$ pipenv install 1password
Creating a virtualenv for this project...
Pipfile: /tmp/tmp.XY4KfI3Or1/Pipfile
Using /usr/bin/python3 (3.8.5) to create virtualenv...
⠹ Creating virtual environment...created virtual environment CPython3.8.5.final.0-64 in 93ms
  creator CPython3Posix(dest=/home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/isme/.local/share/virtualenv)
    added seed packages: pip==21.1.2, setuptools==57.0.0, wheel==0.36.2
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

✔ Successfully created virtual environment! 
Virtualenv location: /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU
Creating a Pipfile for this project...
Installing 1password...
Error:  An error occurred while installing 1password!
Error text: Collecting 1password
  Using cached 1password-0.5.2.tar.gz (11 kB)
Collecting wget>=3.2
  Using cached wget-3.2-py3-none-any.whl
Collecting pyyaml>=5.4
  Using cached PyYAML-5.4.1-cp38-cp38-manylinux1_x86_64.whl (662 kB)
Collecting pycryptodome>=3.9.7
  Using cached pycryptodome-3.10.1-cp35-abi3-manylinux2010_x86_64.whl (1.9 MB)
Collecting pexpect>=4.7.0
  Using cached pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
Collecting ptyprocess>=0.5
  Using cached ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)
Building wheels for collected packages: 1password
  Building wheel for 1password (setup.py): started
  Building wheel for 1password (setup.py): finished with status 'error'
  Running setup.py clean for 1password
Failed to build 1password
Installing collected packages: ptyprocess, wget, pyyaml, pycryptodome, pexpect, 1password
    Running setup.py install for 1password: started
    Running setup.py install for 1password: finished with status 'error'

  ERROR: Command errored out with exit status 1:
   command: /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-eget46_i/1password_cd666bc5e89641a09b31d4f972ab6641/setup.py'"'"'; __file__='"'"'/tmp/pip-install-eget46_i/1password_cd666bc5e89641a09b31d4f972ab6641/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-do9f0zmb
       cwd: /tmp/pip-install-eget46_i/1password_cd666bc5e89641a09b31d4f972ab6641/
  Complete output (38 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/onepassword
  copying onepassword/exceptions.py -> build/lib/onepassword
  copying onepassword/client.py -> build/lib/onepassword
  copying onepassword/utils.py -> build/lib/onepassword
  copying onepassword/__init__.py -> build/lib/onepassword
  warning: build_py: byte-compiling is disabled, skipping.
  
  installing to build/bdist.linux-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.linux-x86_64
  creating build/bdist.linux-x86_64/wheel
  creating build/bdist.linux-x86_64/wheel/onepassword
  copying build/lib/onepassword/exceptions.py -> build/bdist.linux-x86_64/wheel/onepassword
  copying build/lib/onepassword/client.py -> build/bdist.linux-x86_64/wheel/onepassword
  copying build/lib/onepassword/utils.py -> build/bdist.linux-x86_64/wheel/onepassword
  copying build/lib/onepassword/__init__.py -> build/bdist.linux-x86_64/wheel/onepassword
  warning: install_lib: byte-compiling is disabled, skipping.
  
  running install_egg_info
  running egg_info
  writing 1password.egg-info/PKG-INFO
  writing dependency_links to 1password.egg-info/dependency_links.txt
  writing requirements to 1password.egg-info/requires.txt
  writing top-level names to 1password.egg-info/top_level.txt
  reading manifest file '1password.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching 'LICENSE.txt'
  adding license file 'LICENSE'
  writing manifest file '1password.egg-info/SOURCES.txt'
  Copying 1password.egg-info to build/bdist.linux-x86_64/wheel/1password-0.5.2-py3.8.egg-info
  running install_scripts
  error: [Errno 1] Operation not permitted: '/usr/local/bin/'
  ----------------------------------------
  ERROR: Failed building wheel for 1password
    ERROR: Command errored out with exit status 1:
     command: /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-eget46_i/1password_cd666bc5e89641a09b31d4f972ab6641/setup.py'"'"'; __file__='"'"'/tmp/pip-install-eget46_i/1password_cd666bc5e89641a09b31d4f972ab6641/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-j8uwdzmr/install-record.txt --single-version-externally-managed --compile --install-headers /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/include/site/python3.8/1password
         cwd: /tmp/pip-install-eget46_i/1password_cd666bc5e89641a09b31d4f972ab6641/
    Complete output (35 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib
    creating build/lib/onepassword
    copying onepassword/exceptions.py -> build/lib/onepassword
    copying onepassword/client.py -> build/lib/onepassword
    copying onepassword/utils.py -> build/lib/onepassword
    copying onepassword/__init__.py -> build/lib/onepassword
    warning: build_py: byte-compiling is disabled, skipping.
    
    running install_lib
    creating /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/lib/python3.8/site-packages/onepassword
    copying build/lib/onepassword/exceptions.py -> /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/lib/python3.8/site-packages/onepassword
    copying build/lib/onepassword/client.py -> /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/lib/python3.8/site-packages/onepassword
    copying build/lib/onepassword/utils.py -> /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/lib/python3.8/site-packages/onepassword
    copying build/lib/onepassword/__init__.py -> /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/lib/python3.8/site-packages/onepassword
    warning: install_lib: byte-compiling is disabled, skipping.
    
    running install_egg_info
    running egg_info
    writing 1password.egg-info/PKG-INFO
    writing dependency_links to 1password.egg-info/dependency_links.txt
    writing requirements to 1password.egg-info/requires.txt
    writing top-level names to 1password.egg-info/top_level.txt
    reading manifest file '1password.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching 'LICENSE.txt'
    adding license file 'LICENSE'
    writing manifest file '1password.egg-info/SOURCES.txt'
    Copying 1password.egg-info to /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/lib/python3.8/site-packages/1password-0.5.2-py3.8.egg-info
    running install_scripts
    writing list of installed files to '/tmp/pip-record-j8uwdzmr/install-record.txt'
    error: [Errno 1] Operation not permitted: '/usr/local/bin/'
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-eget46_i/1password_cd666bc5e89641a09b31d4f972ab6641/setup.py'"'"'; __file__='"'"'/tmp/pip-install-eget46_i/1password_cd666bc5e89641a09b31d4f972ab6641/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-j8uwdzmr/install-record.txt --single-version-externally-managed --compile --install-headers /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/include/site/python3.8/1password Check the logs for full command output.

✘ Installation Failed 
```

**Comment by dtpryce** [comment]
Yeah looks exactly the same as mine which I can manually fix by installing `op` ... can you try `op --version` and `which op` in your pipenv for me?

**Comment by iainelder** [comment]
> can you try op --version and which op in your pipenv for me?

Like this? When I run those commands in the pipenv, it finds op.

```
$ pipenv run which op
/usr/local/bin/op
$ pipenv run op --version
1.10.3
```

Despite the error from the install command, it appears that the Python code does get copied to the right place.

So I can even partially reproduce the the  [check_install_required](https://github.com/wandera/1password-client/blob/c890b3b1f647fa8620e95d0aecf85c44851ce8aa/install_op.py#L54) function.

It also returns a result that would indicate that installation is not required.

```
$ python -c 'from onepassword.utils import read_bash_return as r; print(r("op --version"))'
1.10.3
```

I haven't figured out a way to properly load the install_op.py script because it's apparently not copied to site-packages.

```
$ tree /home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/lib/python3.8/site-packages/onepassword
/home/isme/.local/share/virtualenvs/tmp.XY4KfI3Or1-dpap2ByU/lib/python3.8/site-packages/onepassword
├── client.py
├── exceptions.py
├── __init__.py
└── utils.py
```

**Comment by iainelder** [comment]
I can download the install_op script directly from Github and run check_install_required in my pipenv. It returns False.

```text
$ wget -q 'https://raw.githubusercontent.com/wandera/1password-client/master/install_op.py'
$ pipenv run python -c "import install_op as i; print(i.check_install_required())"
False
```

**Comment by dtpryce** [comment]
Yeah the python code will get downloaded but just not installed into pip and symlinked etc but not sure myself what is going on now. If the pipenv is pointing to what is already installed then it shouldn't need to run that install_op steps. I will investigate both in my own machine and then with a new pipenv also. 

Thanks for the info, sorry I can't suggest a fix yet. It might end up being an extra like you suggest but must also be backwards compatible.

**Comment by dtpryce** [comment]
Ooo I think I have worked it out ... the if statement could be moved to wrap earlier and this might all be fine. I will check it out tomorrow. Thanks for your patience!

**Comment by iainelder** [comment]
Thanks for your attention! I appreciate you taking the time to investigate this although it's working in your environment.

You had me confused for a minute there when you mentioned the "if", until I realized I was looking one level too deep.

The installation failure log doesn't include a Python stack trace, so I had implicitly assumed it was failing when trying to copy a file into /usr/local/bin.

In fact it fails before running check_install_required command. It fails when it tries to change the permissions of /usr/local/bin.

https://github.com/wandera/1password-client/blob/c890b3b1f647fa8620e95d0aecf85c44851ce8aa/install_op.py#L76

Here's how I reproduced it:

```
$ wget -q 'https://raw.githubusercontent.com/wandera/1password-client/master/install_op.py'
$ pipenv run python -c "import install_op as i; print(i.install_op())"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/tmp/tmp.XY4KfI3Or1/install_op.py", line 76, in install_op
    os.chmod(local_bin, 0o755)
PermissionError: [Errno 1] Operation not permitted: '/usr/local/bin/'
```

So, if I understand you correctly, the install_op command should call check_install_required immediately, and just return immediately if it returns false.

Looking forward to your thoughts tomorrow.

**Comment by dtpryce** [comment]
If you want to try early it should be on this branch https://github.com/wandera/1password-client/tree/pre-installed-op and you can install locally using `pip install -e . ` or `python setup.py install` ... otherwise wait for the new version and test there

**Comment by dtpryce** [comment]
Fixed in #29

**Comment by iainelder** [comment]
Thanks! The installation process now works as expected.
